DXMainClient/Domain/Multiplayer/CnCNet/MapSharer.cs

```diff
@@ -380,20 +373,20 @@ public static string GetMapFileName(string sha1, string mapName)
 
         private static string DownloadMain(string sha1, string myGame, string mapName, out bool success)
         {
-            string customMapsDirectory = ProgramConstants.GamePath + "Maps/Custom/";
+            string customMapsDirectory = SafePath.CombineDirectoryPath(ProgramConstants.GamePath, "Maps", "Custom");
 
             string mapFileName = GetMapFileName(sha1, mapName);
 
-            string destinationFilePath = customMapsDirectory + mapFileName + ".zip";
+            FileInfo destinationFile = SafePath.GetFile(customMapsDirectory, FormattableString.Invariant($"{mapFileName}.zip"));
 
             // This string is up here so we can check that there isn't already a .map file for this download.
             // This prevents the client from crashing when trying to rename the unzipped file to a duplicate filename.
-            string newFilename = customMapsDirectory + mapFileName + ".map";
+            FileInfo newFile = SafePath.GetFile(customMapsDirectory, FormattableString.Invariant($"{mapFileName}.map"));
 
             try
             {
-                if (File.Exists(destinationFilePath)) File.Delete(destinationFilePath);
-                if (File.Exists(newFilename)) File.Delete(newFilename);
+                if (destinationFile.Exists) destinationFile.Delete();
+                if (newFile.Exists) newFile.Delete();
```

Comment:
Can just use your delete if exists and omit try catch.

Reply:
Won't it be very time consuming to collect stack information when creating exceptions?

Reply:
My bad. It doesn't throw an exception when the file is not exist.

See [dotnet docs](https://docs.microsoft.com/dotnet/api/system.io.fileinfo.delete?view=net-6.0#remarks)
![image](https://user-images.githubusercontent.com/42184238/189459980-b1bed4f8-0148-4640-a37e-530b90d77a6e.png)

```suggestion
                try
                {
                    destinationFile.Delete();
                    newFile.Delete();
                } catch { }
```

Reply:
I am talking about RansAckeR's implementation of it in SafePath, not the system one. But yeah even better.